src/app/mod.rs

```diff
@@ -326,10 +327,10 @@ impl Surreal {
 	/// const people = await db.query('SELECT * FROM type::table($table)', { table: 'person' });
 	/// ```
 	pub async fn query(&self, sql: String, bindings: JsValue) -> Result<JsValue, Error> {
-		let bindings: Json = from_value(bindings)?;
-		let mut response = match bindings.is_null() {
-			true => self.db.query(sql).await?,
-			false => self.db.query(sql).bind(bindings).await?,
+		let bindings = json(&from_value::<Json>(bindings)?.to_string()).unwrap_or(Value::None);
```

Comment:
If the `json` function fails to parse the JSON returned here, we want to know about it not fall back to `Value::None`. This might silently ignore bugs in our parser. I would rather we return the error here.